docs/webiny-apps/page-builder/development/creating-iframe-element-plugin.md

```diff
@@ -0,0 +1,197 @@
+ plugin type.
+
+
+```jsx
+import React from "react";
+import styled from "@emotion/styled";
+import {
+    PbEditorPageElementPlugin
+} from "@webiny/app-page-builder/types";
+import { ReactComponent as IFrameIcon } from "./iframe-icon.svg";
+import IFrameEmbed from "./iFrameEmbed";
+
+const PreviewBox = styled("div")({
+    textAlign: "center",
+    height: 50,
+    svg: {
+        height: 50,
+        width: 50
+    }
+});
+export default () => {
+
+    return {
+            name: "pb-editor-page-element-iframe",
+            type: "pb-editor-page-element",
+            elementType: "iframe",
+            toolbar: {
+                title: "iFrame",
+                group: "pb-editor-element-group-media",
+                preview() {
+                    return (
+                        <PreviewBox>
+                            <IFrameIcon />
+                        </PreviewBox>
+                    );
+                }
+            },
+            settings: ["pb-editor-page-element-settings-delete"],
+            onCreate: "open-settings",
+            // Create function creates an element data structure.
+            create(options) {
+                return {
+                    type: "iframe",
+                    elements: [],
+                    data: {
+                        iframe: {
+                            // The URL property will be populated when user enters the URL in the settings dialog
+                            url: "",
+                            height: 370
+                        },
+                        settings: {
+                            horizontalAlign: "center",
+                            margin: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            },
+                            padding: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            }
+                        }
+                    },
+                    ...options
+                };
+            },
+            render(props) {
+                return <IFrameEmbed {...props} />;
+            },
+            renderElementPreview({ width, height }) {
+                return <img style={{ width, height }} alt={"iFrame"} />;
+            }
+        } as PbEditorPageElementPlugin,
+    // Settings dialog comes here
+```
+
+The key properties of the plugin are the `create`, `data`, `settings`, and `render`. They contribute to the state of our plugin. On changing the state, it’s up to the render function to render the component based on the state changes.
+
+Since we are adding a new element in the editor, the type of plugin will be `pb-editor-page-element`. This plugin consists of different properties, one of them is the `toolbar` property that helps us put our element on the media group, using `pb-editor-element-group-media`, as seen in the image below.
```

Comment:
> us put our element on the media group...

Change to: `us put our element into the "Media" group`.

BTW, can you try to switch places here and try putting this "Since we are adding..." sentence before the "The key properties..." ? The "Since we are adding ..." just sounds a bit better as a continuation of the image.  Not sure, give it a try and let us know if it makes it any better.

Reply:
Yes you are right the properties that I mentioned are better in the order of putting the "Since we are adding.." before the "The key properties..."